internal/state/execution.go

```diff
@@ -310,29 +310,33 @@ func (blockExec *BlockExecutor) ApplyBlock(
 	return state, nil
 }
 
-func (blockExec *BlockExecutor) ExtendVote(ctx context.Context, vote *types.Vote) (types.VoteExtension, error) {
+func (blockExec *BlockExecutor) ExtendVote(ctx context.Context, vote *types.Vote) ([]byte, error) {
 	req := abci.RequestExtendVote{
-		Vote: vote.ToProto(),
+		Hash:   vote.BlockID.Hash,
+		Height: vote.Height,
 	}
 
 	resp, err := blockExec.appClient.ExtendVote(ctx, req)
 	if err != nil {
-		return types.VoteExtension{}, err
+		return nil, err
 	}
-	return types.VoteExtensionFromProto(resp.VoteExtension), nil
+	return resp.VoteExtension, nil
 }
 
 func (blockExec *BlockExecutor) VerifyVoteExtension(ctx context.Context, vote *types.Vote) error {
 	req := abci.RequestVerifyVoteExtension{
-		Vote: vote.ToProto(),
+		Hash:             vote.BlockID.Hash,
+		ValidatorAddress: vote.ValidatorAddress,
+		Height:           vote.Height,
+		VoteExtension:    vote.Extension,
```

Comment:
Sanity check here please?

Reply:
Good question. Perhaps we should skip this application call and automatically accept any vote that's missing a vote extension? cc @sergio-mena 

Not sure if I missed this in the spec or if we didn't state it explicitly, but right now I can't imagine any harm coming from automatically accepting in the absence of an extension.

Reply:
Currently, the spec implies that `VerifyVoteExtension` will be called even with empty extension, although it doesn't state this clearly.
I think the best would be to discuss this on Wednesday's ABCI++ meeting. I am not 100% sure what is better from the App's perspective, so would like us to have stakeholders' input before we make a decision.
I just opened in Issue to keep track of this decision. Until then, I guess either solution should do here.

Reply:
Adding on to William's reasoning (that I agree with). Also think about the difference between:
* An empty, signed, vote extension
* No vote extension
These may seem very similar but are different. The latter can be tampered with in the current vote extension model, the former cannot.
As a conclusion, I think that, when in doubt, we need to give the application as much visibility as possible. But, as stated previously, I'd like to have the App developers' input in tomorrow's ABCI++ meeting.

Reply:
From todays `ABCi++` call, the decision was made to always pass the extension to the application, even if it is empty. Applications that do not wish to use the vote extension mechanism must just return the `ACCEPT` `enum` value.